libsel4/tools/rustgenerator.py

```diff
@@ -0,0 +1,615 @@
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+
+from generator import (
+    MAX_MESSAGE_LENGTH,
+    Generator,
+    is_result_struct_required,
+    struct_members,
+    get_parameter_positions
+)
+
+
+class RustGenerator(Generator):
+    @staticmethod
+    def translate_type(name):
+        type_trans = {
+            "int": "isize",
+            "seL4_Uint8": "u8",
+            "seL4_Uint16": "u16",
+            "seL4_Uint32": "u32",
+            "seL4_Uint64": "u64",
+            "seL4_Bool": "u8",
+            "seL4_CapData_t": "seL4_CapData",
+            "seL4_PrioProps_t": "seL4_PrioProps",
+            "seL4_CapRights_t": "seL4_CapRights",
+            "seL4_RISCV_Page_GetAddress_t": "seL4_RISCV_Page_GetAddress",
```

Comment:
It might be useful in the long run to have a way to hide architecture specific types, so they can't get in by quirky accidented eventually.